lib/saluki-io/src/deser/codec/dogstatsd/mod.rs

```diff
@@ -317,6 +337,127 @@ fn parse_dogstatsd<'a>(
+        let (mut remaining, _) = tag("|")(remaining)?;
+        while let Some((chunk, tail)) = split_at_delimiter(remaining, b'|') {
+            if chunk.len() < 2 {
+                break;
+            }
+            match &chunk[..2] {
+                // Timestamp: client-provided timestamp for the event, relative to the Unix epoch, in seconds.
+                event::TIMESTAMP_PREFIX => {
+                    let (_, timestamp) = all_consuming(preceded(tag(event::TIMESTAMP_PREFIX), unix_timestamp))(chunk)?;
+                    maybe_timestamp = Some(timestamp);
+                }
+                // Hostname: client-provided hostname for the host that this event originated from.
+                event::HOSTNAME_PREFIX => {
+                    let (_, hostname) =
+                        all_consuming(preceded(tag(event::HOSTNAME_PREFIX), event_option_value))(chunk)?;
+                    maybe_hostname = Some(hostname.to_string());
+                }
+                // Aggregation key: key to be used to group this event with others that have the same key.
+                event::AGGREGATION_KEY_PREFIX => {
+                    let (_, aggregation_key) =
+                        all_consuming(preceded(tag(event::AGGREGATION_KEY_PREFIX), event_option_value))(chunk)?;
+                    maybe_aggregation_key = Some(aggregation_key.to_string());
+                }
+                // Priority: client-provided priority of the event.
+                event::PRIORITY_PREFIX => {
+                    let (_, priority) =
+                        all_consuming(preceded(tag(event::PRIORITY_PREFIX), event_option_value))(chunk)?;
+                    match priority {
+                        event::PRIORITY_LOW => maybe_priority = Some(saluki_event::eventd::Priority::Low),
+                        _ => maybe_priority = Some(saluki_event::eventd::Priority::Normal),
+                    }
+                }
+                // Source type name: client-provided source type name of the event.
+                event::SOURCE_TYPE_PREFIX => {
+                    let (_, source_type) =
+                        all_consuming(preceded(tag(event::SOURCE_TYPE_PREFIX), event_option_value))(chunk)?;
+                    maybe_source_type = Some(source_type.to_string());
+                }
+                // Alert type: client-provided alert type of the event.
+                event::ALERT_TYPE_PREFIX => {
+                    let (_, alert_type) =
+                        all_consuming(preceded(tag(event::ALERT_TYPE_PREFIX), event_option_value))(chunk)?;
+                    match alert_type {
+                        event::ALERT_TYPE_ERROR => maybe_alert_type = Some(saluki_event::eventd::AlertType::Error),
+                        event::ALERT_TYPE_WARNING => maybe_alert_type = Some(saluki_event::eventd::AlertType::Warning),
+                        event::ALERT_TYPE_SUCCESS => maybe_alert_type = Some(saluki_event::eventd::AlertType::Success),
```

Comment:
One way to clean this up, and just sort of as a general pattern to follow: any time you're trying to parse some structured type from unstructured data (like going from a raw string to `AlertType`), etc... you _generally_ want to colocate that conversion logic with the the type itself.

There's some traits in Rust around conversions, both fallible and infallible, but in our case, we know we're dealing with `Option<T>`, and there's no trait for fallible conversions that return `Option<T>`, so we might just write an inherent method for this, following naming best practices:

```rust
impl AlertType {
    pub fn try_from_string(raw_alert_type: &str) -> Option<Self> {
        ...
    }
}
```

This function would have the same logic that you have here, and this codeblock would then just become `maybe_alert_type = AlertType::try_from_string(alert_type)`. We'd ideally do this for `Priority` as well.